internal/utils/schema.go

```diff
@@ -13,6 +13,6 @@ func ExpandStringSet(set *schema.Set) []string {
 	return strs
 }
 
-func IsKnown(val attr.Value) bool {
-	return !(val.IsNull() || val.IsUnknown())
+func IsKnown[T attr.Value](value T) bool {
```

Comment:
Just curios - what are the benefits of converting the function to generic in this case? Is it just about performance?

Reply:
I don't think there is one. IIRC this function was merged into main after I'd started work on this PR. I'm happy to revert this change if you think that makes sense?

Reply:
Up to you. I don't have any preference. Having said this I came across an [investigation](https://blog.stackademic.com/golang-design-generics-vs-interfaces-how-it-really-works-under-the-hood-c1e5057a7ae6) regarding the topic, still IMO the perf effect if any doesn't make any diff in our case.